scim2-sdk-common/src/main/java/com/unboundid/scim2/common/utils/Parser.java

```diff
@@ -161,6 +177,76 @@ public static Filter parseFilter(final String filterString)
     return readFilter(new StringReader(filterString.trim()), false);
   }
 
+  /**
+   * Add parser options, in order to modify parser behavior within the current
+   * thread.
+   *
+   * <p>NOTE: SCIM server implementations are not guaranteed to support a given option.</p>
+   * <p>NOTE: these should be removed when no longer needed.</p>
+   *
+   * <pre>
+   *   try
+   *   {
+   *     Parser.addOptions(ParserOption.ALLOW_SEMICOLONS_IN_ATTRIBUTE_NAMES();
+   *     performWhateverProcessing();
+   *   }
+   *   finally
+   *   {
+   *     Parser.removeOptions(ParserOption.ALLOW_SEMICOLONS_IN_ATTRIBUTE_NAMES();
+   *   }
+   * </pre>
+   *
+   * @param options  The parser options to be added.
+   */
+  public static void addOptions(final ParserOption... options)
+  {
+    Set<ParserOption> updatedOptions = Parser.threadLocalOptions.get();
+    for (ParserOption option : options)
+    {
+      updatedOptions.add(option);
+    }
+    Parser.threadLocalOptions.set(updatedOptions);
+  }
+
+  /**
+   * Remove parser options, in order to resume default parser behavior within
+   * the current thread.
+   *
+   * @param options  The parser options to be removed.
+   */
+  public static void removeOptions(final ParserOption... options)
+  {
+    Set<ParserOption> updatedOptions = Parser.threadLocalOptions.get();
+    for (ParserOption option : options)
+    {
+      updatedOptions.remove(option);
+    }
+    Parser.threadLocalOptions.set(updatedOptions);
+  }
+
+  /**
+   * Get the current set of parser options.
+   *
+   * @return  The current set of parser options.
+   */
+  public static Set<ParserOption> getOptions()
+  {
+    return Collections.unmodifiableSet(Parser.threadLocalOptions.get());
+  }
+
+  /**
+   * Replace the current set of parser options.
```

Comment:
Like with the other methods, I'd mention that this is just for the current thread.